packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -15,21 +39,46 @@ export const AdvancedSettings = ({
 }: {
   destinationAddress?: string
   onChange: (value?: string) => void
-  error: AdvancedSettingsErrors | null
+  error: DestinationAddressErrors | null
 }) => {
   const {
-    app: { selectedToken }
+    app: { selectedToken, isDepositMode }
   } = useAppState()
+  const { l1, l2 } = useNetworksAndSigners()
   const { isEOA = false, isSmartContractWallet = false } = useAccountType()
 
   const [collapsed, setCollapsed] = useState(true)
+  const [warning, setWarning] = useState<string | null>(null)
 
   useEffect(
     // Show on page load if SC wallet since destination address mandatory
     () => setCollapsed(!isSmartContractWallet),
     [isSmartContractWallet]
   )
 
+  useEffect(() => {
+    // checks if trying to send to a contract address
+    async function getWarning() {
```

Comment:
NIT: For the sake of consistency - I'd keep this as a function alongside `getDestinationAddressError` with the name `getDestinationAddressWarning` and would keep the implementation on similar lines and add comments.

Reply:
updated